encoding/ssz/htrutils.go

```diff
@@ -141,19 +141,21 @@ func WithdrawalSliceRoot(withdrawals []*enginev1.Withdrawal, limit uint64) ([32]
 	return MixInLength(bytesRoot, bytesRootBufRoot), nil
 }
 
-func transactionRoot(tx []byte) ([32]byte, error) {
-	chunkedRoots, err := PackByChunk([][]byte{tx})
+// ByteSliceRoot is a helper func to merkleize an arbitrary List[Byte, N]
+// this func runs Chunkify + MerkleizeVector
+// max length is dividable by 32 ( root length )
+func ByteSliceRoot(slice []byte, maxLength uint64) ([32]byte, error) {
+	chunkedRoots, err := PackByChunk([][]byte{slice})
 	if err != nil {
 		return [32]byte{}, err
 	}
-
-	maxLength := (fieldparams.MaxBytesPerTxLength + 31) / 32
-	bytesRoot, err := BitwiseMerkleize(chunkedRoots, uint64(len(chunkedRoots)), uint64(maxLength))
+	maxRootLength := (maxLength + 31) / 32 // nearest number divisible by root length (32)
```

Comment:
Can we add a comment as to why we are doing `+ 31` and since `32` is the root length would it be better to minimize the magic numbers and have a const for this?

Reply:
> (value + 31)/32
is so that the ending chunk if it isnt fully occupied ( < 32 bytes) is still
correctly addressed as 1 full chunk.

potuz and nishant gave me this info.
i added the comment to represent this, any other opinions on making this clearer?